flink-end-to-end-tests/run-nightly-tests.sh

```diff
@@ -43,121 +43,151 @@ EXIT_CODE=0
 #     printf "\n==============================================================================\n"
 #     printf "Running my fancy nightly end-to-end test\n"
 #     printf "==============================================================================\n"
+#     start_timer
 #     $END_TO_END_DIR/test-scripts/test_something_very_fancy.sh
 #     EXIT_CODE=$?
+#     end_timer
 # fi
 
 
 if [ $EXIT_CODE == 0 ]; then
     printf "\n==============================================================================\n"
     printf "Running HA end-to-end test\n"
     printf "==============================================================================\n"
+    start_timer
     $END_TO_END_DIR/test-scripts/test_ha.sh
     EXIT_CODE=$?
+    end_timer
 fi
 
 if [ $EXIT_CODE == 0 ]; then
   printf "\n==============================================================================\n"
   printf "Running Resuming Savepoint (file, async, no parallelism change) end-to-end test\n"
   printf "==============================================================================\n"
+  start_timer
   STATE_BACKEND_TYPE=file STATE_BACKEND_FILE_ASYNC=true $END_TO_END_DIR/test-scripts/test_resume_savepoint.sh 2 2
   EXIT_CODE=$?
+  end_timer
 fi
 
 if [ $EXIT_CODE == 0 ]; then
   printf "\n==============================================================================\n"
   printf "Running Resuming Savepoint (file, sync, no parallelism change) end-to-end test\n"
   printf "==============================================================================\n"
+  start_timer
   STATE_BACKEND_TYPE=file STATE_BACKEND_FILE_ASYNC=false $END_TO_END_DIR/test-scripts/test_resume_savepoint.sh 2 2
   EXIT_CODE=$?
+  end_timer
 fi
 
 if [ $EXIT_CODE == 0 ]; then
   printf "\n==============================================================================\n"
   printf "Running Resuming Savepoint (file, async, scale up) end-to-end test\n"
   printf "==============================================================================\n"
+  start_timer
   STATE_BACKEND_TYPE=file STATE_BACKEND_FILE_ASYNC=true $END_TO_END_DIR/test-scripts/test_resume_savepoint.sh 2 4
   EXIT_CODE=$?
+  end_timer
 fi
```

Comment:
Perhaps we can refactor this whole block into a common base script for the nightly / pre-commit hook scripts.
The end_timer / start_timer functions should also be located there.